gossip/src/duplicate_shred_handler.rs

```diff
@@ -131,6 +148,20 @@ impl DuplicateShredHandler {
                     shred1.into_payload(),
                     shred2.into_payload(),
                 )?;
+                if let Some(epoch_schedule) = &self.cached_epoch_schedule {
+                    // feature_epoch could only be 0 in tests and new cluster setup.
+                    if self
+                        .enable_gossip_duplicate_proof_ingestion_epoch
+                        .is_some_and(|feature_epoch| {
+                            epoch_schedule.get_epoch(slot) > feature_epoch || feature_epoch == 0
```

Comment:
should this be `>= feature_epoch`, presuming `feature_set.activated_slot()`  returns the first slot where the feature is active

Reply:
we add a 1 epoch delay because `enable_gossip_duplicate_proof_ingestion_epoch` is derived from the root bank which doesn't cross the epoch boundary in a synchronous fashion.

Reply:
> the root bank which doesn't cross the epoch boundary in a synchronous fashion

Hmmm, don't we always activate the feature at least an epoch in advance, so it should be rare that a root bank hasn't been rooted in an epoch?

Reply:
if you don't have the delay, `>= feature_epoch` then you run into the following scenario:
assume epoch boundary is slot 100
receive shred from slot 132. some people have root bank 99 some people have root bank 100
for root bank 99 `feature_set.activated_slot` will be `None` but for root bank 100 `feature_aset.activated_slot` will be `100`

we add the 1 epoch delay under the assumption that at least one super majority root must be made per epoch.

Reply:
Ah yeah, that makes sense